lib/options.go

```diff
@@ -44,6 +45,21 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+type DNSConfig struct {
+	TTL null.String `json:"ttl"`
+}
+
+func (c *DNSConfig) Decode(value string) error {
```

Comment:
hmm instead of a custom `Decode()` function, why not implement the [`TextUnmarshaler` interface](https://golang.org/pkg/encoding/#TextUnmarshaler)?

Reply:
You prefer `TextUnmarshaler` because it's more widely used?

Done in 5ef75377, but I also had to implement `UnmarshalJSON` for some reason.

Reply:
well, we definitely need to be able to put it in the `metadata.json` and back out when making the archive and then running it ;)

Reply:
This is resolved now.

Reply:
Because there is no `Decoder` interface in the Go standard library AFAIK. The convention is that if your type can be encoded as text or decoded from text, you can implement the `ecnoding.TextMarshaler` or `ecnoding.TextUnmarshaler` and different things will now know to work with your type. For example, [envconfig](https://github.com/kelseyhightower/envconfig/#supported-struct-field-types) and even JSON: https://golang.org/pkg/encoding/json/#example__textMarshalJSON

Reply:
Ah right, for some reason I was thinking [`json.Decoder`](https://golang.org/pkg/encoding/json/#Decoder) was an interface. Yeah, better to use an interface in this case for sure.